shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1951,7 +1974,12 @@ - (void)onUserSettingsChanged:(NSNotification*)notification {
 }
 
 - (CGFloat)textScaleFactor {
+#if APPLICATION_EXTENSION_API_ONLY
+  UIContentSizeCategory category =
+      self.mainScreenIfViewLoaded.traitCollection.preferredContentSizeCategory;
```

Comment:
I think you can just use this in all cases since this is all available on iOS 10+, but I didn't test it so I put it in the macro for safety in my PR.